### PR TITLE
Use `default_manager` instead of `objects` in `ModelIndexView`

### LINKED ISSUE #11765: `ModelIndexView` does not work with custom object managers
### Issue Summary

In `ModelIndexView._get_snippet_types`, the `count` is done by accessing `model.objects.all().count()`. This does not work with models that have a default manager that is not `objects`.

When registering a model as a snippet that defines a custom default manager with a name different than `objects` it will throw an exception. In our case:

```
AttributeError: type object 'ParserError' has no attribute 'objects'
(4 additional frame(s) were not displayed)
...
  File "wagtail/admin/auth.py", line 165, in decorated_view
    response = view_func(request, *args, **kwargs)
  File "django/views/generic/base.py", line 98, in view
    self.setup(request, *args, **kwargs)
  File "wagtail/snippets/views/snippets.py", line 85, in setup
    self.snippet_types = self._get_snippet_types()
  File "wagtail/snippets/views/snippets.py", line 88, in _get_snippet_types
    return [
  File "wagtail/snippets/views/snippets.py", line 91, in <listcomp>
    "count": model.objects.all().count(),
```

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. Add a new model that has a manager not named "objects" (see example below)
3. Register that model as a snippet
4. Go to snippets view
5. See error

```py
from django.db import models
from wagtail.snippets.models import register_snippet

class ParserErrorManager(models.Manager):
   pass

@register_snippet
class ParserError(models.Model):
   err_msg = models.CharField(max_length=250)

    errors = ParserErrorManager()
```

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes / no)

### Technical details

-   Python version: 3.10.
-   Django version: 4.2.8.
-   Wagtail version: 5.2.3.
-   Browser version: https://www.whatsmybrowser.org/b/RFN3C

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

### PR DESCRIPTION
**Pull request by jhonatan-lopes**
Fixes #11765 by using the model's [`_default_manager`](https://docs.djangoproject.com/en/5.0/topics/db/managers/#django.db.models.Model._default_manager) to get the count instead of the hardcoded `objects`

### CODE REVIEW

**Comment by jhonatan-lopes** [comment]
I haven't added tests because this seems like a very straightforward fix. Besides, from what I could initially see there are no tests setup for the `ModelIndexView` itself (it seems to be tested as part of `SnippetViewSet`), so it would require me to setup tests for this view, create a new model on `testapp` with a custom default manager, all the works...

If that's necessary though, just let me know

**Review by laymonage — COMMENTED** [review]
Thanks, tests are available here

https://github.com/wagtail/wagtail/blob/04ade6d217b0c8ab3746383e92e14c32e224a7f9/wagtail/snippets/tests/test_snippets.py#L80-L112

You should be able to change one of the snippet test model definitions to trigger the issue, e.g. the `AdvertWithCustomPrimaryKey` https://github.com/wagtail/wagtail/blob/cd50955b4984b156250d1382b2cc9c880779b57c/wagtail/test/testapp/models.py#L946

File: wagtail/test/testapp/models.py [hunk 1]
```diff
@@ -1011,6 +1011,28 @@ class Meta:
 register_snippet(AdvertWithTabbedInterface)
 
 
+class AdvertManager(models.Manager):
+    pass
+
+
+class AdvertWithCustomManager(models.Model):
```
Comment (laymonage) [inline wagtail/test/testapp/models.py:1018]: Thanks, but we don't need a new model for this, let's just reuse an existing one. Adding a new model would add additional overhead as that's yet another migration to run and another table to create on the database.

Comment (jhonatan-lopes) [reply wagtail/test/testapp/models.py:1018]: Will do :+1:

Comment (jhonatan-lopes) [reply wagtail/test/testapp/models.py:1018]: Wouldn't that create a new migration in any case? Adding a new manager to an existing model will need a migration as well, won't it?

Comment (laymonage) [reply wagtail/test/testapp/models.py:1018]: You're right, yep, but it won't create a new database table.

Comment (jhonatan-lopes) [reply wagtail/test/testapp/models.py:1018]: I just tried here locally with the `Advert` model (the simplest one), and besides creating the migration I would also have to change all occurrences of `Advert.objects.foo` in the tests (around 70).

The one with the least amount of test occurrences (2) is `AdvertWithCustomUUIDPrimaryKey`, but I think putting a custom manager on this one hurts the readability of tests.

Please let me know how you'd like to proceed

Comment (laymonage) [reply wagtail/test/testapp/models.py:1018]: My bad, yep. We can go with a simple model without any fields (other than the primary key), and name it something like `CustomManagerModel` or `ModelWithCustomManager` I suppose.

**Review by laymonage — APPROVED** [review]
Thanks! I can confirm this fixes the issue. I've squashed the commits, formatted the migrations, and added the release note. Will merge once CI finishes.

I don't think this is going to be backported to the current releases though, as this issue is not a recent regression. Looking from the code, it has been around since the instance counts were added in 4.0.

**Pull request merged**